tests/backup/backup_helper.go

```diff
@@ -1348,3 +1350,119 @@ func restoreSuccessCheck(restoreName string, orgID string, retryDuration int, re
 	log.Infof("[%s] restored successfully", restoreName)
 	return restoreStatus, nil
 }
+
+// CreateCustomRestoreWithPVCs function can be used to deploy custom deployment with it's PVCs. It cannot be used for any other resource type.
+func CreateCustomRestoreWithPVCs(restoreName string, backupName string, namespaceMapping map[string]string, clusterName string,
+	orgID string, ctx context.Context, storageClassMapping map[string]string, namespace string) (deploymentName string, err error) {
+
+	var bkp *api.BackupObject
+	var bkpUid string
+	var newResources []*api.ResourceInfo
+	var options metav1.ListOptions
+	var deploymentPvcMap = make(map[string][]string)
+	var deploymentNames []string
+	backupDriver := Inst().Backup
+	log.Infof("Getting the UID of the backup needed to be restored")
+	bkpEnumerateReq := &api.BackupEnumerateRequest{
+		OrgId: orgID}
+	curBackups, err := backupDriver.EnumerateBackup(ctx, bkpEnumerateReq)
+	if err != nil {
+		return "", err
+	}
+	for _, bkp = range curBackups.GetBackups() {
+		if bkp.Name == backupName {
+			bkpUid = bkp.Uid
+			break
+		}
+	}
+	deploymentList, err := apps.Instance().ListDeployments(namespace, options)
+	if err != nil {
+		return "", fmt.Errorf("unable to list the deployments with error %v", err)
+	}
+	if len(deploymentList.Items) == 0 {
+		return "", fmt.Errorf("deployment list is null")
+	}
+	deployments := deploymentList.Items
+	for _, deployment := range deployments {
+		var pvcs []string
+		for _, vol := range deployment.Spec.Template.Spec.Volumes {
+			pvcName := vol.PersistentVolumeClaim.ClaimName
+			pvcs = append(pvcs, pvcName)
+		}
+		deploymentPvcMap[deployment.Name] = pvcs
+	}
+	log.Infof("deployment pvc map %v", deploymentPvcMap)
```

Comment:
This can be a huge list. No need to print this out

Reply:
Done

Reply:
This is not yet fixed